File: app/src/main/java/com/games/memorygame/model/game/LevelFactory.java
package com.games.memorygame.model.game;


public class LevelFactory {

    public static Level getLevel(LevelType type) {
        switch (type) {
            case EASY:
                return new Level(3, 2, 40, 10);
            case NORMAL:
                return new Level(4, 3, 80, 20);
            case HARD:
                return new Level(5, 4, 120, 30);
            default:
                return new Level(3, 2, 40, 10);
        }
    }

    private LevelFactory() {
    }
}

File: app/src/main/java/com/games/memorygame/model/game/LevelFactory.kt
package com.games.memorygame.model.game


object LevelFactory {

    fun getLevel(type: LevelType): Level {
        return when (type) {
            LevelType.EASY -> Level(3, 2, 40, 10)
            LevelType.NORMAL -> Level(4, 3, 80, 20)
            LevelType.HARD -> Level(5, 4, 120, 30)
        }
    }
}
